TBRM cluster: Various spec discrepancies + alchemy

ksperling-apple opened the pull request:
A number of discrepancies between the spec and the SDK were discovered for the TBRM cluster:

- Get{Active,Pending}DatasetRequest should handle absence of these datasets by returning success with an empty octstr, but the current implementation returned a NotFound status. Test steps added to TC-TBRM-2.2 / TC-TBRM-2.3 and existing steps updated to use Timed Invoke where applicable.
- Generating the cluster XML via Alchemy (#39602, thanks @tersal) revealed set Set{Active,Pending}DatasetRequest did not have the Timed Invoke requirement from the spec reflected in the XML. Test steps added to TC-TBRM-2.3.
- The CASE session requirement for Set{Active,Pending}DatasetRequest was correctly implemented in the SDK, but not tested. Test steps added to TC-TBRM-2.4 for this test gap.

#### Testing

Updated TC-TBRM-2.{2,3,4} to test these changes and correctly reflect the spec.
Corresponding test plan PR: https://github.com/CHIP-Specifications/chip-test-plans/pull/5251

ksperling-apple committed:
TBRM cluster: Handle absence of datasets correctly

The spec mandates a success response with an empty / null octstr for this, but
the current implementation returned a NotFound status.

Update TC-TBRM-2.2 to fill in this test gap.

bzbarsky-apple commented on src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp:
```diff
@@ -67,10 +67,13 @@ Status ServerInstance::HandleGetDatasetRequest(CommandHandlerInterface::HandlerC
+        dataset.Init(ByteSpan());
+        return Status::Success;
     }
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, Status::Failure);
```
```suggestion
    VerifyOrReturnValue(err == CHIP_NO_ERROR, StatusIB(err).mStatus);
```
might be better, in case the error encodes a status already.

ksperling-apple replied:
Changed it to `return StatusIB(err).mStatus;` since that also handles the success case.

ksperling-apple committed:
Address review comments

tersal committed:
Generate ThreadBorderRouter cluster with Alchemy

ksperling-apple committed:
Update tests for Set{Active,Pending}DatasetRequest for timed invoke

- Test server responds with NEEDS_TIMED_INTERACTION without timed invoke
- Use timed invoke for all other usages of these requests

ksperling-apple committed:
Test CASE enforcement for Set{Active,Pending}DatasetRequest

ksperling-apple committed:
Align test step numbers with test plan

As per https://github.com/CHIP-Specifications/chip-test-plans/pull/5251

ksperling-apple committed:
Disable new negative timed-invoke tests with darwin-framework-tool
